File: Coding Ninjas/Moderate/Container With Most Water.py
def maxArea( height ):
    ans = m = 0
    n = len( height ) - 1
    
    while m < n:
        ans = max( min( height[m], height[n] ) * ( n - m ), ans )
        
        if height[m] > height[n]:
            n -= 1
        else:
            m += 1
            
    return ans

File: Coding Ninjas/Moderate/Container With Most Water.java
public class Solution
{
    
    public static int maxArea
    ( int[] height )
    {
        int i = 0;
        int j = height.length - 1;
        int ans = 0;
        
        while ( i < j )
        {
            ans = Math.max( Math.min( height[i], height[j] ) * ( j - i ), ans );

            if ( height[i] > height[j] )
            {
                j--;
            }
            else
            {
                i++;
            }
        }

        return ans;
    }

}
